rasa/core/policies/rule_policy.py

```diff
@@ -571,29 +650,20 @@ def _find_contradicting_rules(
             )
 
         logger.debug("Found no contradicting rules.")
+        all_rules = (
+            set(self._rules_sources.keys())
+            | self._default_sources()
+            | self._handling_loop_sources(domain)
+        )
+        # set is not json serializable, so convert to list
+        return list(all_rules - rules_used_in_stories)
```

Comment:
before it was even worse, it was double usage of `_check_prediction`. Would renaming `_find_contradicting_rules` help?

Reply:
What would you suggest?

I don't think it would help much. In my opinion this mixing multiple things which make the code hard to understand:
* finding rules which are also in stories
* finding conflicts

Reply:
I think it's totally fine it does it in one go over training trackers

Reply:
maybe some more generic name like analyze rules or smth?

Reply:
> maybe some more generic name like analyze rules or smth?

Ok 👍🏻 In combination with a good docstring that could work. I'd also rename / split `_run_prediction_on_trackers` to get rid of the `collect_sources` flag (kinda doing [this](https://martinfowler.com/bliki/FlagArgument.html) refactoring)